addons/point_of_sale/models/pos_session.py

```diff
@@ -2124,6 +2124,12 @@ def get_pos_ui_account_fiscal_positions_by_ids(self, fp_ids):
             fiscal_position['fiscal_position_taxes_by_id'] = {tax_id: fiscal_position_by_id[tax_id] for tax_id in fiscal_position['tax_ids']}
 
         return fps
+    def cash_drawer_open_log(self, employee, action):
+        partner_id = self.env['hr.employee'].browse(employee).work_contact_id.id
+        self.message_post(body='Cash drawer opened ({ACTION})'.format(ACTION=action), author_id=partner_id)
+    def cash_drawer_close_log(self, employee, action):
+        partner_id = self.env['hr.employee'].browse(employee).work_contact_id.id
+        self.message_post(body='Action cancelled ({ACTION})'.format(ACTION=action), author_id=partner_id)
```

Comment:
```suggestion
def log_employee_message(self, employee, action, message_type):
    partner_id = self.env['hr.employee'].browse(employee).work_contact_id.id

    if message_type == 'action_cancelled':
        body = 'Action cancelled ({ACTION})'.format(ACTION=action)
    elif message_type == 'cash_drawer_action':
        body = 'Cash drawer opened ({ACTION})'.format(ACTION=action)

    self.message_post(body=body, author_id=partner_id)
```
Maybe do it like this? This is more generic and will allow us to add logs if needed without having to create other functions?